src/utils/duration.jsx

```diff
@@ -0,0 +1,157 @@
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React from 'react';
+
+export const formatDuration = microseconds => {
+    const usec = Math.floor(microseconds);
+    const u = `${usec % 1000}`;
+
+    if (usec < 1000) return `${u}\u00B5s`;
+
+    const t = new Date(Math.floor(usec / 1000));
+    const z = `${t.getUTCMilliseconds()}`;
+    if (usec < 10000) return `${z}.${u.padStart(3, '0')}ms`;
+    if (usec < 100000) return `${z}.${u.padStart(3, '0').substr(0, 2)}ms`;
+    if (usec < 1000000) return `${z}.${u.padStart(3, '0').substr(0, 1)}ms`;
+
+    const s = `${t.getUTCSeconds()}`;
+    if (usec < 10000000) return `${s}.${z.padStart(3, '0')}s`;
+    if (usec < 60000000) return `${s}.${z.padStart(3, '0').substr(0, 2)}s`;
+
+    const m = `${t.getUTCMinutes()}`;
+    if (usec < 600000000)
+        return `${m}:${s.padStart(2, '0')}.${z.padStart(3, '0').substr(0, 1)}m`;
+    if (usec < 3600000000) return `${m}:${s.padStart(2, '0')}m`;
+
+    const h = `${t.getUTCHours()}`;
+    if (usec < 86400000000)
+        return `${h}:${m.padStart(2, '0')}:${s.padStart(2, '0')}h`;
+
+    const d = Math.floor(usec / 86400000000);
+    return `${d}d ${h.padStart(2, '0')}:${m.padStart(2, '0')}h`;
+};
+
+export const formatDurationHTML = microseconds => {
+    if (Number.isNaN(microseconds)) return null;
+    const usec = Math.floor(microseconds);
+    const u = `${usec % 1000}`;
+
+    if (usec < 1000)
+        return (
+            <div className="value">
+                {u}
+                <span className="unit">{'\u00B5s'}</span>
+            </div>
+        );
+    const t = new Date(Math.floor(usec / 1000));
+    const z = `${t.getUTCMilliseconds()}`;
+
+    if (usec < 10000)
+        return (
+            <div className="value">
+                {`${z}.${u.padStart(3, '0')}`}
+                <span className="unit">ms</span>
+            </div>
+        );
```

Comment:
Would be nice to extract the html part into its own function, such as:
```jsx
if (usec < 1000) return formatHtml(`${z}.${u.padStart(3, '0')}`, 'ms')
```

```jsx
const formatHtml = (duration, value) => (
    <div className="value">
         {duration}
        <span className="unit">{value}</span>
    </div>
)
```

This would yield less repetition and make it easier if we should ever want to update the html.

Reply:
This utility would be a perfect match for a test no? :D

Reply:
Yes it would be best to test. Html part is not the same for all cases, maybe we refactor it later.

Reply:
It is the same for all but the last. And we all know that refactoring later will never happen :P